compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -348,6 +348,65 @@ void RecordMinMax::profileRawData(const std::string &mode, const std::string &in
   update_quantparam(getObserver(), mode, min_percentile, max_percentile);
 }
 
+std::vector<std::vector<std::vector<char>>>
+RecordMinMax::importH5Data(const std::string &input_data_path)
```

Comment:
Q) will this read all the files in the path? what if there are lots of files that exceeds memory?

Reply:
@BalyshevArtem It would be better to catch `std::bad_alloc` to handle the case.

```
  catch (const std::bad_alloc &e)
  {
    throw std::runtime_error("Out of memory during h5 data load.");
  }
```

Reply:
@BalyshevArtem , I and @seanshpark had offline talk. We're concerned about excessive memory consumption. Could you add a code to check the size of input data and reject if the whole data size is too large ? We think 1 GB of the max byte size is acceptable.

Reducing the memory usage of parallel recording would be a future development item, which is necessary to enable parallel recording in our public toolchain (one-quantize).

Reply:
Done